src/common/utils/resolve_reference.py

```diff
@@ -10,6 +10,31 @@
 from storage.data_source_class import DataSource
 
 
+def split_reference(reference: str) -> list[str]:
+    """Splits a reference into it's string components
+    e.g. "$123.content[1]" -> ["$123", ".content", "[1]"]"""
```

Comment:
Is it wise to introduce yet another split reference format, here with "."? Should we not agree on one format internally in dmt and stick to it?

Reply:
The problem is that I need the "string parts". Not a list of RefereceItems.

I tried to reuse the "_next_reference_part()", but it was returning all kinds of stuff like empty string, single dots and so on. Also it removes the prefix, like ".", "[" and "(". 

But I agree, this should not be duplicated. Can see if we can make this work